VideoWeb/VideoWeb/Helpers/ParticipantsUpdatedEventNotifier.cs

```diff
@@ -0,0 +1,55 @@
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading.Tasks;
+using VideoWeb.Common.Models;
+using VideoWeb.Contract.Responses;
+using VideoWeb.EventHub.Handlers.Core;
+using VideoWeb.EventHub.Models;
+using VideoWeb.Helpers.Interfaces;
+using VideoWeb.Mappings;
+using EventType = VideoWeb.EventHub.Enums.EventType;
+
+namespace VideoWeb.Helpers
+{
+    public class ParticipantsUpdatedEventNotifier: IParticipantsUpdatedEventNotifier
+    {     
+        private readonly IEventHandlerFactory _eventHandlerFactory;
+        private readonly IMapperFactory _mapperFactory;
+        private readonly ILogger<ParticipantsUpdatedEventNotifier> _logger;
+
+        public ParticipantsUpdatedEventNotifier(IEventHandlerFactory eventHandlerFactory, IMapperFactory mapperFactory, ILogger<ParticipantsUpdatedEventNotifier> logger)
+        {
+            _eventHandlerFactory = eventHandlerFactory;
+            _mapperFactory = mapperFactory;
+            _logger = logger;
+        }
+
+        public async Task PushParticipantsUpdatedEvent(Conference conference)
+        {
+            var participantsToResponseMapper = _mapperFactory.Get<Participant, Conference, ParticipantResponse>();
+            CallbackEvent callbackEvent = new CallbackEvent()
+            {
+                ConferenceId = conference.Id,
+                EventType = EventType.ParticipantsUpdated,
+                TimeStampUtc = DateTime.UtcNow,
+                Participants = conference.Participants.Select(participant => participantsToResponseMapper.Map(participant, conference)).ToList()
+            };
+
+            _logger.LogTrace($"Publishing event to UI: {JsonSerializer.Serialize(callbackEvent)}");
+            await PublishEventToUi(callbackEvent);
+        }
+
+        private Task PublishEventToUi(CallbackEvent callbackEvent)
+        {
+            if (callbackEvent == null)
+            {
+                return Task.CompletedTask;
+            }
+
+            var handler = _eventHandlerFactory.Get(callbackEvent.EventType);
+            return handler.HandleAsync(callbackEvent);
```

Comment:
is this async just by name or actually an async method, if its the latter we should make this as async method and use await here

Reply:
Honestly I'm not totally clear on why it is this way. This came from internal events controller as I didn't want to duplicate this code in QuickLinks controller. And this is the same as in VideoEvents controller so maybe that should be put somewhere central itself. But this is how all the events are currently handled. Perhaps it is something that could be looked at but it wouldn't just be for this event.

Reply:
I'm not 100% sure on this but I would have thought this is ok because we are returning a task and it is awaited elsewhere by the code. In this case on line 41. Although I think this await should be in the controller rather than in the this class.